docs/develop/python/message-passing.mdx

```diff
@@ -27,155 +27,179 @@ tags:
   - dynamic-handlers
 ---
 
-This page shows how to do the following:
+Here is an example Workflow Definition featuring Query, Signal, and Update handlers, and an Update validator.
+You can view this as a runnable sample at [hello/hello_message_passing.py](https://github.com/temporalio/samples-python/blob/dan/dan/docs_message_passing.py).
 
-- [Develop with Signals](#signals)
-- [Develop with Queries](#queries)
-- [Develop with Updates](#updates)
-- [Dynamic Handler](#dynamic-handler)
+```python
```

Comment:
Thanks for calling this out -- indeed, we don't love having the huge sample at the top, it breaks with our docs conventions and I don't think it necessarily makes the page easier to read. In this case, I don't think the scaffolding adds very much, since it's relatively straightforward to explain Signals, Queries, and Updates individually -- no need to have them all in one block like that.

Reply:
I know there was a desire to get docs' eyes on this ASAP so porting to other languages can begin, and I would say that this is the primary change I'd like to see before ports; other edits will be relatively minor.

Reply:
Hi @axfelix, I've pushed a change that breaks apart the message handler code snippets and sends them to their respective sections. With the simpler structure and reduction in prose content it's still very easy to find the code that way, so I think both solutions have their merits, and don't dislike the broken out form. What we were investigating with the other form was creating a code sample that allows some developers to learn quickly from code, without navigating prose sections.